### PR TITLE
set miner/ validator cut percentage

### PR DESCRIPTION
**Pull request by mz-Datura**
## Description

This PR implements subnet-specific cut distribution for RAO emissions.
### Changes:
Localizes owner cut per subnet (netuid specific)
Implements RAO emission distribution between:
Miner cut
Validator cut
Burn cut
Ensures all cuts sum to 100% (u16::MAX) per subnet
### Validation:
Added test cases to verify cut calculations
Ensures proper distribution across all subnets
Maintains total cut sum constraint (100%)
Technical Details:
Each subnet maintains independent cut values
Burn cut automatically adjusts to maintain 100% total
All arithmetic operations use safe math to prevent overflow
### Testing:
Test cases verify:
Initial cut distribution
Cut modifications maintain 100% total
Subnet-specific cut independence
Proper error handling for invalid values

## Related Issue(s)

- Closes #[issue number]

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please describe):

## Breaking Change

If this PR introduces a breaking change, please provide a detailed description of the impact and the migration path for existing applications.

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have run `cargo fmt` and `cargo clippy` to ensure my code is formatted and linted correctly
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

## Screenshots (if applicable)

Please include any relevant screenshots or GIFs that demonstrate the changes made.

## Additional Notes

Please provide any additional information or context that may be helpful for reviewers.

### CODE REVIEW

**Comment by surcyf123** [comment]
open this into devnet-ready @mz-Datura

File: pallets/subtensor/src/lib.rs [hunk 1]
```diff
@@ -434,6 +434,19 @@ pub mod pallet {
     pub fn DefaultSubnetOwnerCut<T: Config>() -> u16 {
         T::InitialSubnetOwnerCut::get()
     }
+
+    #[pallet::type_value]
+    /// Default value for subnet minter cut.
```
Comment (cuteolaf) [inline pallets/subtensor/src/lib.rs]:
```suggestion
    /// Default value for subnet miner cut.
```

File: pallets/subtensor/src/lib.rs [hunk 2]
```diff
@@ -434,6 +434,19 @@ pub mod pallet {
...
+    pub fn DefaultSubnetMinterCut<T: Config>() -> u16 {
```
Comment (cuteolaf) [inline pallets/subtensor/src/lib.rs]:
```suggestion
    pub fn DefaultSubnetMinerCut<T: Config>() -> u16 {
```

File: pallets/subtensor/src/lib.rs [hunk 3]
```diff
@@ -434,6 +434,19 @@ pub mod pallet {
...
+        T::InitialSubnetMinterCut::get()
```
Comment (cuteolaf) [inline pallets/subtensor/src/lib.rs]:
```suggestion
        T::InitialSubnetMinerCut::get()
```

File: pallets/subtensor/src/lib.rs [hunk 4]
```diff
@@ -884,6 +897,14 @@ pub mod pallet {
     #[pallet::storage]
     /// ITEM( subnet_owner_cut )
     pub type SubnetOwnerCut<T> = StorageValue<_, u16, ValueQuery, DefaultSubnetOwnerCut<T>>;
+    
+    #[pallet::storage]
+    /// ITEM( subnet_minter_cut )
+    pub type SubnetMinterCut<T> = StorageValue<_, u16, ValueQuery, DefaultSubnetMinterCut<T>>;
```
Comment (cuteolaf) [inline pallets/subtensor/src/lib.rs]:
```suggestion
    /// ITEM( subnet_miner_cut )
    pub type SubnetMinerCut<T> = StorageValue<_, u16, ValueQuery, DefaultSubnetMinerCut<T>>;
```

File: pallets/subtensor/src/macros/config.rs [hunk 5]
```diff
@@ -174,6 +174,13 @@ mod config {
         /// Initial network subnet cut.
         #[pallet::constant]
         type InitialSubnetOwnerCut: Get<u16>;
+        /// Initial subnet minter cut.
+        #[pallet::constant]
+        type InitialSubnetMinterCut: Get<u16>;
```
Comment (cuteolaf) [inline pallets/subtensor/src/macros/config.rs]:
```suggestion
        /// Initial subnet miner cut.
        #[pallet::constant]
        type InitialSubnetMinerCut: Get<u16>;
```

File: pallets/subtensor/src/macros/events.rs [hunk 6]
```diff
@@ -111,6 +111,10 @@ mod events {
         Faucet(T::AccountId, u64),
         /// the subnet owner cut is set.
         SubnetOwnerCutSet(u16),
+        /// the subnet minter cut is set.
+        SubnetMinterCutSet(u16),
```
Comment (cuteolaf) [inline pallets/subtensor/src/macros/events.rs]:
```suggestion
        /// the subnet miner cut is set.
        SubnetMinerCutSet(u16),
```

File: pallets/subtensor/src/utils/misc.rs [hunk 7]
```diff
@@ -611,6 +611,26 @@ impl<T: Config> Pallet<T> {
...
     }
 
+    pub fn set_subnet_minter_cut(subnet_minter_cut: u16) {
+        SubnetMinterCut::<T>::set(subnet_minter_cut);
+        Self::deposit_event(Event::SubnetMinterCutSet(subnet_minter_cut));
+    }
+
+    pub fn get_subnet_minter_cut() -> u16 {
+        SubnetMinterCut::<T>::get()
+    }
```
Comment (cuteolaf) [inline pallets/subtensor/src/utils/misc.rs]:
```suggestion
    pub fn set_subnet_miner_cut(subnet_minter_cut: u16) {
        SubnetMinterCut::<T>::set(subnet_miner_cut);
        Self::deposit_event(Event::SubnetMinerCutSet(subnet_miner_cut));
    }

    pub fn get_subnet_miner_cut() -> u16 {
        SubnetMinerCut::<T>::get()
    }
```

File: runtime/src/lib.rs [hunk 8]
```diff
@@ -952,6 +952,8 @@ parameter_types! {
     pub const SubtensorInitialMinAllowedUids: u16 = 128;
     pub const SubtensorInitialMinLockCost: u64 = 1_000_000_000_000; // 1000 TAO
     pub const SubtensorInitialSubnetOwnerCut: u16 = 11_796; // 18 percent
+    pub const SubtensorInitialSubnetMinterCut: u16 = 11_796; // 18 percent
```
Comment (cuteolaf) [inline runtime/src/lib.rs]:
```suggestion
    pub const SubtensorInitialSubnetMinerCut: u16 = 11_796; // 18 percent
```

File: runtime/src/lib.rs [hunk 9]
```diff
@@ -1018,6 +1020,8 @@ impl pallet_subtensor::Config for Runtime {
     type InitialNetworkMinLockCost = SubtensorInitialMinLockCost;
     type InitialNetworkLockReductionInterval = SubtensorInitialNetworkLockReductionInterval;
     type InitialSubnetOwnerCut = SubtensorInitialSubnetOwnerCut;
+    type InitialSubnetMinterCut = SubtensorInitialSubnetMinterCut;
```
Comment (cuteolaf) [inline runtime/src/lib.rs]:
```suggestion
    type InitialSubnetMinerCut = SubtensorInitialSubnetMinerCut;
```

**Review by cuteolaf — CHANGES_REQUESTED** [review]

**Comment by cuteolaf** [comment]
What are the new storage items for?
Did you open this PR based on a discussion or some kind of approval from the OTF team?
The logic doesn't make much sense.

**Comment by cuteolaf** [comment]
Could you add the required fields?
e.g. description, type of change, etc.

**Comment by mz-Datura** [comment]
> Could you add the required fields? e.g. description, type of change, etc.

name is correct, please don't change item name. thanks

**Pull request closed without merging**